### PR TITLE
Adding the inspect command

### PR DESCRIPTION
**Pull request by ktsakalozos**
First pass in introspection script.

Fixes: https://github.com/ubuntu/microk8s/issues/37

### CODE REVIEW

File: scripts/inspect.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+INSPECT_DUMP=${SNAP_DATA}/inspection-report
+RETURN_CODE=0
+
+function print_help {
+  # Print the help message
+  printf -- 'This script will inspect your microk8s installation. It will report any issue it may find,\n';
```
Comment (tvansteenburgh) [inline scripts/inspect.sh]: "report any issues it finds,"

File: scripts/inspect.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  printf -- 'and prepare tarball with logs and traces you can attach on issues you open against\n';
```
Comment (tvansteenburgh) [inline scripts/inspect.sh]: "and create a tarball of logs and traces which can be attached to an issue filed against"

File: scripts/inspect.sh [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  if iptables -L | grep FORWARD | grep DROP &> /dev/null
+  then
+      printf -- '\033[0;33m WARNING: \033[0m IPtables FORWARD policy is DROP. '
+      printf -- 'Consider enabling traffic forwarding with: sudo iptables -P FORWARD ACCEPT \n'
+  fi
+
+  ufw=$(ufw status)
+  if echo $ufw | grep "Status: active" &> /dev/null && ! echo $ufw | grep cbr0 &> /dev/null
+  then
+      printf -- '\033[0;33m WARNING: \033[0m Firewall is enebled. Consider allowing pod traffic '
```
Comment (tvansteenburgh) [inline scripts/inspect.sh]: s/enebled/enabled/

File: scripts/inspect.sh [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+check_service "snap.microk8s.daemon-docker"
+check_service "snap.microk8s.daemon-apiserver"
+check_service "snap.microk8s.daemon-proxy"
+check_service "snap.microk8s.daemon-kubelet"
+check_service "snap.microk8s.daemon-scheduler"
+check_service "snap.microk8s.daemon-controller-manager"
+check_service "snap.microk8s.daemon-etcd"
+store_args
+
+printf -- 'Inspect AppArmor configuration\n'
```
Comment (tvansteenburgh) [inline scripts/inspect.sh]: s/Inspect/Inspecting/  (to be consistent with above)

File: scripts/inspect.sh [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+check_apparmor
+
+printf -- 'Gather system info\n'
```
Comment (tvansteenburgh) [inline scripts/inspect.sh]: Gathering

File: scripts/inspect.sh (hunk 1 above)
Comment (ktsakalozos) [reply scripts/inspect.sh]: Done

File: scripts/inspect.sh (hunk 2 above)
Comment (ktsakalozos) [reply scripts/inspect.sh]: Easy

File: scripts/inspect.sh (hunk 3 above)
Comment (ktsakalozos) [reply scripts/inspect.sh]: Wasy

File: scripts/inspect.sh (hunk 4 above)
Comment (ktsakalozos) [reply scripts/inspect.sh]: Yeap

File: scripts/inspect.sh (hunk 5 above)
Comment (ktsakalozos) [reply scripts/inspect.sh]: Also yeap

**Comment by ktsakalozos** [comment]
Thank you, merging this

**Pull request merged**